pkg/volume/iscsi/iscsi_util_test.go

```diff
@@ -69,3 +71,39 @@ func TestExtractPortalAndIqn(t *testing.T) {
 		t.Errorf("extractPortalAndIqn: got %v %s %s", err, portal, iqn)
 	}
 }
+
+func fakeOsStat(devicePath string) (fi os.FileInfo, err error) {
+	var cmd os.FileInfo
+	return cmd, nil
+}
+
+func fakeFilepathGlob(devicePath string) (globs []string, err error) {
+	return []string{devicePath}, nil
+}
+
+func TestWaitForPathToExist(t *testing.T) {
+	devicePath := []string{"/dev/disk/by-path/ip-127.0.0.1:3260-iqn.2014-12.com.example:test.tgt00-lun-0",
+		"/dev/disk/by-path/pci-0000:00:00.0-ip-127.0.0.1:3260-iqn.2014-12.com.example:test.tgt00-lun-0"}
+	osStat = fakeOsStat
+	defer func() { osStat = os.Stat }()
```

Comment:
because it would call `waitForPathInternal` instead of `WaitForPath` ? I think that is still better. As long as `waitForPath` is a thin function which only sets up the default arguments and calls `waitForPathInternal`. Using variables like these for mocknig purposes makes the code harder to read and debug. Imagine you missed one of the deferred calls and where trying to debug why the following test is failing for example